### PR TITLE
no need to assume we are dealing with deadbeats

### LINKED ISSUE #586: upon cold app start, you are a deadbeat is shown briefly
Upon starting the app in the not-yet-logged-in state, the gallery is briefly shown with the "you are a deadbeat" message. Shortly thereafter the sign in screen is shown. Upon signing in successfully, the gallery is shown again and again (or still) shows the "you are a deadbeat" message for some moments before the contents are reloaded, assuming the account is not marked as deadbeat server side.

Simulator Screenshot - iPhone 12 Pro

app: 6.8 (58)

It seems the issue was introduced with #583 where the default of showing user as deadbeat was not immediately overwritten when the user was not yet logged in.

### PR DESCRIPTION
**Pull request by krugerk**
## Summary
App started off claiming user is a deadbeat.
The state resulted from recent changes in #583 where the deadbeat view was hidden or not only when a user was logged in _and_ the default state being to show the deadbeat view.

*For UI changes including screenshots of before and after is great.*

#### before
Simulator Screenshot - iPhone 12 Pro

#### after
Simulator Screenshot - iPhone 12 Pro - 2024-12-30 at 14 33 48

## Validation
signed out, started app anew, signed in
in simulator

Fixes #586

### CODE REVIEW

File: BeeSwift/Gallery/GalleryViewController.swift [hunk 1]
```diff
@@ -115,6 +115,7 @@ class GalleryViewController: UIViewController, UICollectionViewDelegateFlowLayou
             make.left.equalTo(10)
             make.right.equalTo(-10)
         }
+        self.deadbeatView.isHidden = true
```
Comment (krugerk) [inline BeeSwift/Gallery/GalleryViewController.swift:118]: show it when we know better

File: BeeSwift/Gallery/GalleryViewController.swift [hunk 2]
```diff
@@ -289,8 +290,8 @@ class GalleryViewController: UIViewController, UICollectionViewDelegateFlowLayou
     
 
     func updateDeadbeatVisibility() {
-        guard let user = currentUserManager.user(context: viewContext) else { return }
-        self.deadbeatView.isHidden = !user.deadbeat
+        let isKnownDeadbeat = currentUserManager.user(context: viewContext)?.deadbeat == true
```
Comment (theospears) [inline BeeSwift/Gallery/GalleryViewController.swift:293]:
```suggestion
        let isKnownDeadbeat = currentUserManager.user(context: viewContext)?.deadbeat ?? false
```
This is the same, right?

**Review by theospears — APPROVED** [review]
Good catch, thanks for fixing this 🏅

File: BeeSwift/Gallery/GalleryViewController.swift (hunk 2 above)
Comment (krugerk) [reply BeeSwift/Gallery/GalleryViewController.swift:293]: Effectively, yes

**Pull request merged**